api/src/main/java/org/openmrs/module/kenyaemr/reporting/cohort/definition/evaluator/VMMCRegisterCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.reporting.cohort.definition.evaluator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Cohort;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.kenyaemr.reporting.cohort.definition.VMMCRegisterCohortDefinition;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+
+/**
+ * Evaluator for patients for VMMC Register
+ */
+@Handler(supports = {VMMCRegisterCohortDefinition.class})
+public class VMMCRegisterCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+
+	protected static final Log log = LogFactory.getLog(VMMCRegisterCohortDefinition.class);
+
+	@Autowired
+	EvaluationService evaluationService;
+
+
+	/**
+	 * @see CohortDefinitionEvaluator#evaluate(CohortDefinition, EvaluationContext)
+	 */
+	@Override
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) throws EvaluationException {
+
+		VMMCRegisterCohortDefinition cohortddef = (VMMCRegisterCohortDefinition) cohortDefinition;
+
+		Cohort newCohort = new Cohort();
+
+		Date startDate = (Date) context.getParameterValue("startDate");
+		Date endDate = (Date) context.getParameterValue("endDate");
+
+		String qry = "SELECT t.patient_id from kenyaemr_etl.etl_vmmc_enrolment t inner join kenyaemr_etl.etl_patient_demographics d on t.patient_id = d.patient_id and date(t.visit_date) BETWEEN date(:startDate) AND date(:endDate) GROUP BY t.patient_id;";
```

Comment:
Is there a possibility to have multiple enrollments for a patient within a given reporting period?